lib/chef/provider/launchd.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Author:: Mike Dodge (<mikedodge04@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/provider/file'
+
+class Chef
+  class Provider
+    class Launchd < Chef::Provider::File
+      provides :launchd
+
+      def initialize(new_resource, run_context)
+        @content_class = Chef::Provider::Launchd::Content
+        super
+      end
+
+      def load_current_resource
+        @current_resource = Chef::Resource::Launchd.new(@new_resource.name)
+        super
+      end
+
+      def define_resource_requirements
+        super
+        requirements.assert(:create, :create_if_missing, :touch) do |a|
+          service_type = @new_resource.service_type.to_s
+          a.assertion { !['system', 'user'].include?(service_type) }
+          a.failure_message(
+            Chef::Exceptions::UnsupportedAction,
```

Comment:
I'd save UnsupportedAction for just when an *action* is provided that the provider doesn't support. Probably best to use `Chef::Exceptions::Service` here.